server/bin/pbench-index.py

```diff
@@ -423,23 +468,29 @@ def main(options, name):
                     rename_tb_link(
                         tb, Path(controller_path, f"{linkerrdest}.1"), idxctx.logger,
                     )
-                elif tb_res in (2, 3):
+                elif tb_res.value in (2, 3):
```

Comment:
We need to think about what to call it; but it'd be cleaner to encapsulate this check (along with the >=4 or <12 below) into the metadata. Perhaps what we need is something other than just "ignorable". I don't fully understand the real semantic intent of this code ... which is exactly why I don't like it, and would like to replace it with something that's more clear as well as cleanly encapsulated.

2 and 3 are "configuration data" errors (missing or bad), but it's not clear we ever actually use those for `tb_res` in the first place, nor what it would mean since `tb_res` deals with the tarball.

I still can't imagine the purpose of checking `tb_res >=4 or res <= 11`, and that bothers me; encapsulating the decisions would make this much more clear. Perhaps Pete and/or Nick can recall the original intent, and we can design something that makes sense, is clear to a reader, and maintainable???

Reply:
Not sure exactly how to answer this, but perhaps by looking at the original code it will make sense.

In the original code, prior to line 406, the variable `tb_res` was set to an integer value by the `try`/`except`/`else` block from lines 306 - 372.   Then the `if` / `elif` / `else` block here from lines 406 - 448 attempts to decide what actions to take based on the numeric error code value.  This `if` block attempts to deal with the error codes in numerical order:  `0` success (lines 407 - 416), `1` indexing error (lines 418 - 425), `2` & `3` logic bomb! (lines 427 - 430; this should not happen because the `try`/`except`/`else` block from lines 306 - 372 does not assign `tb_res` to a value of `2` or `3`), `4` thru `11` quietly move the tar ball to a error destination without logging anything (due to the fact that those errors are logged elsewhere), `else` all other errors logged and moved to the default error destination.

The danger with encapsulation here is that we'll be adding a lot of mechanism to the code which is all inline and not separated across modules.  So we want to be careful we don't over engineer this fix.